core/frontend/src/extension/ExtensionHost.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Extensions
+ */
+
+import { IModelApp } from "../IModelApp";
+import type { ToolAdmin } from "../tools/ToolAdmin";
+import type { AccuSnap } from "../AccuSnap";
+import type { NotificationManager } from "../NotificationManager";
+import type { ViewManager } from "../ViewManager";
+import type { ElementLocateManager } from "../ElementLocateManager";
+import type { Localization } from "@itwin/core-common";
+import type { RenderSystem } from "../render/RenderSystem";
+
+/**
+ * Subset of IModelApp exposed to Extensions
+ * @beta
+ */
+export class ExtensionHost {
```

Comment:
We should definitely explore that to see if it's make things a bit easier. However, I think as we progress with the API we'll find we're adding things that are in addition to the existing iModelApp support or changing quite a bit so in that case it may be a bit too restrictive.

Reply:
made a note for us to revisit this and see if we'd want to switch. @aruniverse I'm going to close this one out.